GetMessage/handler.ts

```diff
@@ -252,19 +303,33 @@ export function GetMessageHandler(
     }
     const maybeMessageStatus = errorOrMaybeMessageStatus.right;
 
-    const returnedMessage = {
-      message,
-      notification: pipe(notificationStatuses, O.toUndefined),
-      // we do not return the status date-time
-      status: pipe(
-        maybeMessageStatus,
-        O.map(messageStatus => messageStatus.status),
-        // when the message has been received but a MessageStatus
-        // does not exist yet, the message is considered to be
-        // in the ACCEPTED state (not yet stored in the inbox)
-        O.getOrElse(() => MessageStatusValueEnum.ACCEPTED)
-      )
-    };
+    const returnedMessage = pipe(
+      {
+        message,
+        notification: pipe(notificationStatuses, O.toUndefined),
+        // we do not return the status date-time
+        status: pipe(
+          maybeMessageStatus,
+          O.map(messageStatus => messageStatus.status),
+          // when the message has been received but a MessageStatus
+          // does not exist yet, the message is considered to be
+          // in the ACCEPTED state (not yet stored in the inbox)
+          O.getOrElse(() => MessageStatusValueEnum.ACCEPTED)
+        )
+      },
+      // Enrich message info with advanced properties if user is allowed to read them
+      messageWithoutAdvacedProperties =>
+        b.fold(
+          () => messageWithoutAdvacedProperties,
+          () => ({
+            ...messageWithoutAdvacedProperties,
+            read_status: b.fold(
+              () => ReadStatusEnum.UNAVAILABLE,
+              () => getReadStatusForService(maybeMessageStatus)
+            )(canReadMessageReadStatus(auth.subscriptionId))
+          })
+        )(canReadAdvancedMessageInfo(message, auth.groups))
```

Comment:
```suggestion
        pipe(
          messageWithoutAdvacedProperties,
          E.fromPredicate(
            () => canReadAdvancedMessageInfo(message, auth.groups),
            () => messageWithoutAdvacedProperties
          ),
          E.map(() => canReadMessageReadStatus(auth.subscriptionId)),
          E.map(
            b.fold(
              () => ReadStatusEnum.UNAVAILABLE,
              () => getReadStatusForService(maybeMessageStatus)
            )
          ),
          E.map(read_status => ({
            ...messageWithoutAdvacedProperties,
            read_status
          })),
          E.toUnion
        )
```
we could flatten the nested fold using an either, but i'm not sure witch one is more readable

Reply:
The one with less row, in my opinion :D